### PR TITLE
Add brotli support to CMake build

### PR DESCRIPTION
**Pull request by JosiahWI**
This looks for the brotlicommon and brotlienc libraries. If they are found, it links both the brotlicommon and brotlienc libraries to the compress plugin.

### CODE REVIEW

**Comment by JosiahWI** [comment]
[[ci approve autest]]

**Comment by JosiahWI** [comment]
[approve ci autest]

File: include/tscore/ink_config.h.cmake.in [hunk 1]
```diff
@@ -104,6 +104,7 @@ const int DEFAULT_STACKSIZE = @DEFAULT_STACK_SIZE@;
 #define TS_MAX_HOST_NAME_LEN @TS_MAX_HOST_NAME_LEN@
 
 /* Feature Flags */
+#cmakedefine01 HAVE_BROTLI_ENCODE_H
```
Comment (cmcfarlen) [inline include/tscore/ink_config.h.cmake.in:107]: Move this up with the other HAVE_ and make it just `#cmakedefine`

Comment (JosiahWI) [reply include/tscore/ink_config.h.cmake.in:107]: There are preprocessor directives that require it to be 0 or 1.

**Comment by bneradt** [comment]
[approve ci centos]

File: include/tscore/ink_config.h.cmake.in [hunk 2]
```diff
@@ -32,6 +32,7 @@
 #cmakedefine BUILD_GROUP "@BUILD_GROUP@"
 #define BUILD_NUMBER "@BUILD_NUMBER@"
 
+#cmakedefine HAVE_BROTLI_ENCODE_H @HAVE_BROTLI_ENCODE_H@
```
Comment (cmcfarlen) [inline include/tscore/ink_config.h.cmake.in:35]: Lets just use `1` here like the rest of these

**Review by cmcfarlen — CHANGES_REQUESTED** [review]
Just use 1 as the value for consistency

**Comment by JosiahWI** [comment]
[approve ci cmake]

**Review by cmcfarlen — APPROVED** [review]

**Pull request merged**